inc/class_parser.php

```diff
@@ -650,18 +650,12 @@ function parse_badwords($message, $options=array())
 					$badword['replacement'] = "*****";
 				}
 
-				if($badword['regex'])
```

Comment:
The whole PR works as expected, however, I'm not sure this is the intended behavior. IMHO, a filter with Regex option turned off should replace things without using regexes; so, if you specify `whatever123*` and Regex is off, it should replace the word literally.

Reply:
Briefly tested that, and it doesn’t work. Even with regex turned off, it also replaces words containing the filtered word, which is wrong.

Reply:
I honestly am not getting the issue. Can you please state steps with what intended and what is happening?

Reply:
Well, turns out I was testing in the wrong branch. It works as expected.

Reply:
Ha ha, thanks for the merge :D